packages/core/src/lib/sign-in.ts

```diff
@@ -98,3 +101,60 @@ export const signInWithPhoneAndPasscode = async (
 
   await assignSignInResult(ctx, provider, id);
 };
+
+// TODO: change this after frontend is ready.
+// Should combian baseUrl(domain) from database with a 'callback' endpoint.
+const connectorRedirectUrl = 'https://logto.dev/callback';
+
+export const assignRedirectUrlForSocial = async (
+  ctx: WithUserLogContext<Context>,
+  connectorId: string,
+  state: string
+) => {
+  const connector = await getSocialConnectorInstanceById(connectorId);
+  assertThat(connector.connector?.enabled, 'connector.not_enabled');
+  const redirectTo = await connector.getAuthorizationUri(connectorRedirectUrl, state);
+  ctx.body = { redirectTo };
+};
+
+const getConnector = async (connectorId: string) => {
+  try {
+    return await getSocialConnectorInstanceById(connectorId);
+  } catch (error: unknown) {
+    // Throw a new error with status 422 when connector not found.
+    if (error instanceof RequestError && error.code === 'entity.not_found') {
```

Comment:
To be more specific, use `connector.not_found` instead of `entity.not_found`? A `connector.not_found` error description is defined in this PR.